core/src/main/scala/com/salesforce/op/stages/impl/regression/RegressionModelSelector.scala

```diff
@@ -195,8 +196,13 @@ case object RegressionModelSelector {
   ): ModelSelector[ModelType, EstimatorType] = {
     val modelStrings = modelTypesToUse.map(_.entryName)
     val modelsToUse =
-      if (modelsAndParameters == defaultModelsAndParams || modelTypesToUse != modelNames) modelsAndParameters
+    // if no models are specified use the defaults and filter by the named models to use
+      if (modelsAndParameters.isEmpty) defaultModelsAndParams
         .filter{ case (e, p) => modelStrings.contains(e.getClass.getSimpleName) }
+      // if models to use has been specified and the models have been specified filter the models by the names
+      else if (modelTypesToUse != modelNames) modelsAndParameters
+        .filter{ case (e, p) => modelStrings.contains(e.getClass.getSimpleName) }
+      // else just use the specified models
```

Comment:
Oh, the third time the same code. Should be fixed.
Also, it needs tests.

Reply:
this has tests in the interface for all three constructor objects